zookeeper/source/zk_nn_client.cc

```diff
@@ -1421,27 +1448,65 @@ bool ZkNnClient::add_block(const std::string &file_path,
   return true;
 }
 
-bool ZkNnClient::add_block_group(const std::string &fileName,
+bool ZkNnClient::add_block_group(const std::string &filePath,
                      u_int64_t &block_group_id,
                      std::vector<std::string> &dataNodes,
-                     std::vector<u_int64_t> &storageBlockIDs,
+                     std::vector<char> &blockIndices,
                      uint32_t total_num_storage_blocks) {
-    FileZNode znode_data;
-    read_file_znode(znode_data, fileName);
+  FileZNode znode_data;
+  read_file_znode(znode_data, filePath);
+  std::vector<u_int64_t> storage_block_ids;
+  // Generate the block group id and storage block ids.
+  block_group_id = generate_block_group_id();
+  for (u_int64_t i = 0; i < total_num_storage_blocks; i++) {
+    storage_block_ids.push_back(generate_storage_block_id(
+        block_group_id, i));
+  }
+
+  // Log them for debugging purposes.
+  LOG(INFO) << "Generated block group id " << block_group_id << "\n";
+  for (auto storageBlockID : blockIndices)
+      LOG(INFO) << "Generated storage block id " << storageBlockID << "\n";
+
+  // TODO(nate): find_data_node_for_block has some other weird logic
+  // baked into it.
+  // TODO(nate): giving it a block_group_id probably makes no sense.
+  std::vector<std::string> tempDataNodes;
+  for (int i = 0; i < total_num_storage_blocks; i++) {
+    find_datanode_for_block(
+        tempDataNodes, storage_block_ids[i],
+        1, true, znode_data.blocksize);
+    dataNodes.push_back(tempDataNodes[0]);
+    tempDataNodes.clear();
+  }
 
-    // Generate the block group id and storage block ids.
-    block_group_id = generate_block_group_id();
-    for (u_int64_t i = 0; i < total_num_storage_blocks; i++) {
-        storageBlockIDs.push_back(generate_storage_block_id(block_group_id, i));
-    }
+  std::vector<std::shared_ptr<ZooOp>> ops;
+  auto block_location_op = zk->build_create_op(
+      get_block_metadata_path(block_group_id),
+      ZKWrapper::EMPTY_VECTOR);
+  ops.push_back(block_location_op);
```

Comment:
It would be good if we had some logging of each op you add to a multi-op so it could be debugged later.

Reply:
(I.e. if the multi-op fails later on )